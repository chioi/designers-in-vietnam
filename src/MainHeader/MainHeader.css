.title-image {
  max-width: 428px;
  width: 100%;
}

.header-link {
  position: relative;
}

.header-link span {
  font-size: var(--font-header-icon);
}

@media screen and (min-width: 480px) {
  .header-link {
    position: absolute;
    right: 0;
    top: 50%;
    -webkit-transform: translateY(-50%);
    -moz-transform: translateY(-50%);
    -ms-transform: translateY(-50%);
    -o-transform: translateY(-50%);
    transform: translateY(-50%);
  }
}
