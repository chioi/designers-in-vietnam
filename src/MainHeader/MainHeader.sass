@import "../style/variables"
@import "../style/mixins/space"
@import "../style/mixins/typography"
@include padding-placeholders($space)
@include fonts($font-sizes)

.title-image
  max-width: 428px
  width: 100%

.header-link
  @extend %padding-vertical-regular
  @extend %padding-horizontal-small
  float: right

.header-link span
  @extend %font-big

.main-navigation
  display: inline-block
  min-width: 320px

.main-header-block
  @extend %padding-vertical-small
  @extend %padding-horizontal-regular
  float: right

@media screen and (min-width: 1280px)
  .logo-container
    min-width: 604px

@media screen and (min-width: 1440px)
  .main-navigation
    min-width: unset

