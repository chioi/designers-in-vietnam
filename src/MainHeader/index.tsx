import { Link } from "@reach/router";
import React from "react";
import Title from "../assets/title.svg";
import "./MainHeader.css";

const MainHeader = () => {
  return (
    <header className="margin-bottom-regular position-relative">
      <Link to="/">
        <img alt="Designers of Vietnam" src={Title} className="title-image" />
      </Link>
      <Link to="/about" className="header-link margin-right-large">
        <span>ğŸ‡»ğŸ‡³</span>
      </Link>
      <a className="header-link" href={process.env.REACT_APP_FORM_URL}>
        <span>ğŸ›µ</span>
      </a>
    </header>
  );
};

export default MainHeader;
