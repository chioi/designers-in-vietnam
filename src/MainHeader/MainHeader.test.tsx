import React from "react";
import {create, ReactTestRenderer} from "react-test-renderer";
import MainHeader from "./index";

describe("gets created", () => {
  let component: ReactTestRenderer;

  beforeAll(() => {
    component = create(<MainHeader />);
  });

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
