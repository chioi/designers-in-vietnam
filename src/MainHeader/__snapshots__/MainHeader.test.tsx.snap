// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets created matches the snapshot 1`] = `
<header
  className="margin-bottom-regular position-relative"
>
  <img
    alt="Designers of Vietnam"
    className="title-image"
    src="title.svg"
  />
  <a
    className="header-link margin-right-large"
    href="/about"
    onClick={[Function]}
  >
    <span>
      ğŸ‡»ğŸ‡³
    </span>
  </a>
  <a
    className="header-link"
  >
    <span>
      ğŸ›µ
    </span>
  </a>
</header>
`;
